src/cursor_on_target.py

```diff
@@ -0,0 +1,183 @@
+    for most local network environments.
+"""
+def setup_multicast_socket(multicast_group, port):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+    sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, 32)
+    return sock
+
+"""
+    Send a CoT message over a multicast UDP socket with error handling and retries.
+
+    Parameters:
+    sock (socket.socket): The UDP socket to use for sending.
+    message (str): The CoT XML message to send.
+    multicast_group (str): The IP address of the multicast group.
+    port (int): The UDP port number to use.
+    max_retries (int, optional): Maximum number of send attempts. Defaults to 3.
+
+    Returns:
+    bool: True if the message was sent successfully, False otherwise.
+
+    Raises:
+    socket.error: If there's a persistent network error after all retries.
+
+    Note:
+    This function implements a simple retry mechanism for handling transient
+    network errors. It will attempt to send the message up to 'max_retries'
+    times before giving up. All errors are logged for debugging purposes.
+    """
+def send_cot_message(sock, message, multicast_group, port, max_retries=3):
+    retries = 0
+    while retries < max_retries:
+        try:
+            sock.sendto(message, (multicast_group, port))
+            logging.info(f"CoT message sent successfully to {multicast_group}:{port}")
+            return True
+        except socket.error as e:
+            retries += 1
+            logging.warning(f"Attempt {retries}/{max_retries} failed. Error: {str(e)}")
+            if retries == max_retries:
+                logging.error(f"Failed to send CoT message after {max_retries} attempts")
+                raise  # Re-raise the last exception
+    return False
+
+"""
+    Create a CoT message and send it over a multicast network.
+
+    This function combines the creation and sending of a CoT message into a single call.
+
+    Parameters:
+    lat (float): Latitude of the target in decimal degrees.
+    lon (float): Longitude of the target in decimal degrees.
+    alt (float): Altitude of the target in meters above WGS84 ellipsoid.
+    ce (float, optional): Circular error of the position estimate in meters. Defaults to 15.0.
+    le (float, optional): Linear error of the altitude estimate in meters. Defaults to 20.0.
+    multicast_group (str, optional): The IP address of the multicast group. Defaults to '239.2.3.1'.
+    port (int, optional): The UDP port number to use. Defaults to 6969.
+
+    Returns:
+    bool: True if the message was sent successfully, False otherwise.
+
+    Note:
+    The default ce and le values are conservative estimates and should be replaced
+    with more accurate values when available from sensor data or calculations.
+    This function will attempt to send the message multiple times in case of network errors.
+"""
+def create_and_send_cot(lat, lon, alt, ce=7.5, le=15.0, multicast_group='239.2.3.1', port=6969):
```

Comment:
In the latest version of OpenAthenaAndroid, we use [a fixed estimate of vertical error](https://github.com/Theta-Limited/OpenAthenaAndroid/blob/5ec06bd1a3b29d1f7a357ef3659fc84dbaf3ea81/app/src/main/java/com/openathena/CursorOnTargetSender.java#L55) (le) at 5.9 meters. This is used with the slant angle to estimate the circular error as the side which completes a right triangle with the slant edge via [this formula here](https://github.com/Theta-Limited/OpenAthenaAndroid/blob/5ec06bd1a3b29d1f7a357ef3659fc84dbaf3ea81/app/src/main/java/com/openathena/CursorOnTargetSender.java#L110).

This is highly preferable as it gives the operator some estimation as to the target calculation accuracy (can be very important depending on use case)